### PR TITLE
Fixing Issues with Async Upstream

### PR DESCRIPTION
**Pull request by leogdion**
### Motivation

Fixing Issues with Building in Swift 6 for Linux

### Modifications

* Fix Issues with Upstream Iterator

### Result

Compatability with Swift 6 on Linux

### Test Plan

TBD

### CODE REVIEW

**Comment by leogdion** [comment]
Apologies @czechboy0 - Fix is pushed.

File: Sources/OpenAPIRuntime/Multipart/MultipartFramesToRawPartsSequence.swift [hunk 1]
```diff
@@ -342,8 +342,17 @@ extension MultipartFramesToRawPartsSequence {
...
-                let frame = try await upstream.next()
+                if #available(macOS 15, iOS 18.0, tvOS 18.0, watchOS 11.0, macCatalyst 18.0, visionOS 2.0, *) {
+                    #if compiler(>=6.0)
+                    frame = try await upstream.next(isolation: self)
+                    #else
+                    frame = try await upstream.next()
+                    #endif
+                } else {
+                    frame = try await upstream.next()
+                }
```
Comment (czechboy0) [inline Sources/OpenAPIRuntime/Multipart/MultipartFramesToRawPartsSequence.swift:355]: While we're here, would you mind creating an extension on `AsyncIteratorProtocol` that adds a utility method for this dance, and call it from these two places you're updating in the PR? It'd take the actor as a parameter.

**Comment by czechboy0** [comment]
@swift-server-bot test this please

**Comment by czechboy0** [comment]
@leogdion Apologies, I didn't realize the utility method strategy won't really work on older toolchains. Can you please revert it and I'll kick off CI again?

**Comment by czechboy0** [comment]
@swift-server-bot test this please

**Comment by czechboy0** [comment]
@leogdion I think if you put the `#available(` into the `#if` compiler check, hopefully the platform should be known and it should fix the CI?

**Comment by leogdion** [comment]
> @leogdion I think if you put the `#available(` into the `#if` compiler check, hopefully the platform should be known and it should fix the CI?

Fixed

**Comment by leogdion** [comment]
@swift-server-bot test this please

**Comment by czechboy0** [comment]
@swift-server-bot test this please

**Review by czechboy0 — APPROVED** [review]
Thanks!

**Pull request merged**